src/nfts/NftGallery.tsx

```diff
@@ -0,0 +1,139 @@
+
+  return (
+    <SafeAreaView testID="NftGallery" style={styles.safeAreaContainer} edges={['top']}>
+      <DrawerTopBar middleElement={<Text>{t('nftGallery.title')}</Text>} />
+      <ScrollView
+        contentContainerStyle={styles.contentContainer}
+        style={styles.scrollContainer}
+        refreshControl={
+          <RefreshControl
+            refreshing={loading}
+            onRefresh={onRefresh}
+            tintColor={colors.greenBrand}
+            colors={[colors.greenBrand]}
+            style={{ backgroundColor: colors.light }}
+            testID="NftGallery/RefreshControl"
+          />
+        }
+      >
+        {error && <NftsLoadError testID="NftGallery/NftsLoadErrorScreen" />}
+        {!error && !loading && (
+          <View style={styles.galleryContainer}>
+            {nfts.length === 0 ? (
+              <Text style={styles.noNfts}>{t('nftGallery.noNfts')}</Text>
+            ) : (
+              nfts.map((nft, idx) => (
+                <Touchable
+                  key={`${nft.contractAddress}-${nft.tokenId}`}
+                  onPress={() => navigate(Screens.NftsInfoCarousel, { nfts: [nft] })}
+                  // Styles for two columns with single items aligned to the left
+                  style={[
+                    styles.touchableIcon,
+                    idx % 2 === 0
+                      ? { marginRight: Spacing.Smallest8 } &&
+                        idx === nfts.length - 1 &&
+                        styles.lastImage
+                      : { marginLeft: Spacing.Smallest8 },
+                  ]}
+                >
+                  <NftImage
+                    nft={nft}
+                    testID="NftGallery/NftImage"
+                    width={imageSize}
+                    height={imageSize}
+                    ErrorComponent={
+                      <View style={styles.errorView}>
+                        <ImageErrorIcon color="#C93717" />
+                      </View>
+                    }
+                    origin={NftOrigin.NftGallery}
+                    borderRadius={Spacing.Smallest8}
+                  />
+                </Touchable>
+              ))
+            )}
+          </View>
+        )}
+      </ScrollView>
+    </SafeAreaView>
+  )
+}
+
+const styles = StyleSheet.create({
+  contentContainer: {
+    flexGrow: 1,
+    padding: Spacing.Regular16,
+    alignItems: 'flex-start',
+  },
+  errorView: {
+    width: imageSize,
+    height: imageSize,
+    flex: 1,
+    alignItems: 'center',
+    justifyContent: 'center',
+    backgroundColor: colors.gray2,
+    borderRadius: Spacing.Regular16,
+  },
+  galleryContainer: {
+    alignContent: 'space-around',
+    justifyContent: 'space-evenly',
+    flexDirection: 'row',
+    flexWrap: 'wrap',
+  },
+  lastImage: {
+    marginRight: variables.width / 2 - Spacing.Smallest8,
+  },
+  noNfts: {
+    ...fontStyles.regular,
+    color: colors.gray3,
+    textAlign: 'center',
+    marginTop: '32%',
```

Comment:
nit: why not use a `Spacing` constant instead?

Reply:
I don't think we have a spacing constant that is big enough. We could add one for '32%' as there are several places it's added as a margin / padding.

Reply:
There's some similar logic in `SwapExecuteScreen.tsx` and `NftsLoadError.tsx`. Using a precent was done as opposed to something like `Spacing.Large32` as it adjusts better for various screen sizes.